sklearn/model_selection/_search.py

```diff
@@ -320,24 +324,61 @@ def fit_grid_point(X, y, estimator, parameters, train, test, scorer,
     return score, parameters, n_samples_test
 
 
+def _check_param_dist(param_dist):
+    """Validate param_dist and return the unique parameters"""
+    parameter_names = set()
+
+    if not isinstance(param_dist, dict):
+        # XXX We do not support list of param_dist dicts for ParameterSampler
+        raise ValueError("The param_dist must be of type dict. %r is not"
+                         % (param_dist,))
+
+    for v in param_dist.values():
+        if not isinstance(v, (list, tuple, np.ndarray, rv_continuous,
```

Comment:
This is not at all appropriate. I'm not sure if it's tested, but we should be accepting any other type of list, or sets; and anything with a callable method `rvs`. Stop adding arbitrary constraints that might break people's code. And more importantly:
Please, Raghav, ***FOCUS***!
I'd really love to see this PR merged, but if you keep drifting from your task, and making more functions that need testing, and more lines to review, you are throwing away reviewers' time/patience/interest. If you're going to be a perfectionist, perfect one thing.